sbin/sync-files.sh

```diff
@@ -28,52 +30,95 @@ cd "${td}"
 
 echo "Getting bucket credentials"
 
-S3CMD="s3cmd $(lpass show --sync=now --notes 9022891142845286058 | jq -r '.AccessKey | "--access_key=\(.AccessKeyId) --secret_key=\(.SecretAccessKey)"')"
+export AWS_ACCESS_KEY_ID="$(lpass show --sync=now --notes 9022891142845286058 | jq -r '.AccessKey | .AccessKeyId')"
+export AWS_SECRET_ACCESS_KEY="$(lpass show --sync=now --notes 9022891142845286058 | jq -r '.AccessKey | .SecretAccessKey')"
 
 echo "Syncing contents of ${BUCKET} to $(pwd)."
-${S3CMD} sync --check-md5 ${BUCKET} ./
+aws s3 sync ${BUCKET} .
 
-FILES=($(ls))
+t=$(mktemp -d)
+pipe=${t}/comms
+trap "rm -f ${pipe}; pkill -P $$" EXIT
+trap "" SIGWINCH
+
+if [[ ! -p ${pipe} ]]; then
+  mkfifo ${pipe}
+fi
+
+ensureFile() {
+  local f="${1}"
+  local u=""
 
-echo "Ensuring that we have the right versions of all files specified in files.json"
-for f in $(< "${jf}" jq -r 'keys[]'); do
   if [ ! -e "${f}" ]; then
     u="$(< "${jf}" jq -r '."'${f}'".URL')"
-    echo "Downloading: ${u}"
-    curl -J -o "${f}" -L --retry 15 --retry-delay 2 $u
+    curl -s -J -o "${f}" -L --retry 15 --retry-delay 2 $u 2>&1
   fi
 
-  s="$(< "${jf}" jq -r '."'${f}'".SHA')"
-  if [ ${#s} -eq 40 ]; then
-    sf="$(shasum "${f}" | cut -d \  -f 1)"
+  local sk="$(< "${jf}" jq -r '."'${f}'".SHA' 2>&1)"
```

Comment:
I find bash kind of cryptic at the best of times, and that combined with the short filenames like `sk`, `jf`, `u` and `sf` here, make it difficult for me to follow the logic

Reply:
sk == sha known
jf == json file
sf == sha file
u == url

Reply:
But yeah. I can rename them.

Reply:
none of that was obvious to me 😅